server/models/profiles.model.js

```diff
@@ -0,0 +1,179 @@
+const mongoose = require("mongoose");
+const httpStatus = require("http-status");
+const async = require("async");
+const bcrypt = require("bcryptjs");
+const APIError = require("../APIError");
+const { env } = require("../vars");
+
+const ProfileSchema = mongoose.Schema(
+  {
+    email: {
+      type: String,
+      match: /^\S+@\S+\.\S+$/,
+      required: true,
+      unique: true,
+      trim: true,
+      lowercase: true
+    },
+    password: {
+      type: String,
+      required: true,
+      minlength: 6,
+      maxlength: 128
+    },
+    image_url: {
+      type: String,
+      trim: true
+    },
+    display_name: {
+      type: String,
+      trim: true
+    },
+    about: {
+      type: String
+    },
+    location: {
+      type: String
+    },
+    website: {
+      type: String,
+      trim: true
+    },
+    phone: {
+      type: String
+    }
+  },
+  {
+    timestamps: true,
+    toJSON: { virtuals: true }
+  }
+);
+
+ProfileSchema.pre("save", async function save(next) {
+  try {
+    if (!this.isModified("password")) return next();
+
+    const rounds = env === "test" ? 1 : 10;
+
+    const hash = await bcrypt.hash(this.password, rounds);
+    this.password = hash;
+
+    return next();
+  } catch (error) {
+    return next(error);
+  }
+});
+
+ProfileSchema.method({
+  transform() {
+    const transformed = {};
+    const fields = [
+      "id",
+      "display_name",
+      "email",
+      "image_url",
+      "phone",
+      "about",
+      "location",
+      "website",
+      "createdAt"
```

Comment:
mixing cases? now you're killing me.  I'll give you a dollar to write a transform method to make the API spit out camel so no one has to know that mongo exists back there and I don't have to convert on the client (because you know I will).

Reply:
I will consider your request, fine sir; but I enjoy giving you the "sads". ;)